### ISSUE TITLE
Air terminal magic not applied to all terminals

### ISSUE THREAD

**Issue by macumber** [NREL/OpenStudio#2022]
From Elizabeth Gillmor: I had always liked that when you set up one of these on an airloop and then added more zones to your loop, that the air terminals, their settings, and loop connections would be magically reproduced for all the subsequent terminals.  However it does not happen with the 2 or 4-pipe induction unit object or chilled beam object.  One has to go manually click on every object to attach these loops. (or spend the time to go write a measure which is not happening right now) That's not exclusive to 1.10 but it is an unpleasant surprise when you are expecting all of them to act the same and they don't.

**Comment by jmarrec** [comment]
I can reproduce this issue in OS 2.2.1. The loop connections aren't preserved with a `AirTerminalSingleDuctConstantVolumeFourPipeInduction`, yet they are with a HW `AirTerminalSingleDuctVAVReheat`

Not sure I understand why, the clone methods looks very similar

https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/model/AirTerminalSingleDuctConstantVolumeFourPipeInduction.cpp#L282

```c++
  ModelObject AirTerminalSingleDuctConstantVolumeFourPipeInduction_Impl::clone(Model model) const
  {
    AirTerminalSingleDuctConstantVolumeFourPipeInduction airTerminalCVFourPipeInductionClone = StraightComponent_Impl::clone(model).cast<AirTerminalSingleDuctConstantVolumeFourPipeInduction>();

    if( boost::optional<HVACComponent> coolingCoil = this->coolingCoil() ) {
      HVACComponent coilCoolingClone = coolingCoil->clone(model).cast<HVACComponent>();
      airTerminalCVFourPipeInductionClone.setCoolingCoil(coilCoolingClone);
    }

    HVACComponent coilHeatingClone = this->heatingCoil().clone(model).cast<HVACComponent>();
    airTerminalCVFourPipeInductionClone.setHeatingCoil(coilHeatingClone);

    return airTerminalCVFourPipeInductionClone;
  }
```

https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/model/AirTerminalSingleDuctVAVReheat.cpp#L279

```c++
  ModelObject AirTerminalSingleDuctVAVReheat_Impl::clone(Model model) const
  {
    AirTerminalSingleDuctVAVReheat modelObjectClone = StraightComponent_Impl::clone(model).cast<AirTerminalSingleDuctVAVReheat>();
    HVACComponent coil = this->reheatCoil();
    HVACComponent coilClone = coil.clone(model).cast<HVACComponent>();
    modelObjectClone.setReheatCoil(coilClone);
    return modelObjectClone;
}
```

**Comment by jmarrec** [comment]
Ok, I just now realized that the connection is done in OS App for AirTerminalSingleDuctVAVReheat, but if you call atu.clone(model), it isn't.

Tracking it down is hard, (had to follow the openstudio_lib Qt stuff), but I'm pretty sure ultimately it calls AirLoopHVAC::addBranchForZone(Impl), which ends up calling `setPlantForAirTerminal`

https://github.com/jmarrec/OpenStudio/blob/develop/openstudiocore/src/model/AirLoopHVAC.cpp#L457

This relies on `containingHVACComponent` returns true. Which is doesn't for the FourPipe one. I'll follow this route, see if it fixes it completely or not.

I'm not 100% sure why atu.clone (or straightcomponent.clone) doesn't automatically connect the loops too (perhaps that's for fear of having incomplete objects?)

```
include OpenStudio::Model

OpenStudio::Logger.instance.standardOutLogger.enable
OpenStudio::Logger.instance.standardOutLogger.setLogLevel(OpenStudio::Debug)

m = Model.new

pl = PlantLoop.new(m)

# Create a AirTerminalSingleDuctVAVReheat with a HW reheatCoil
atu_single_hc = CoilHeatingWater.new(m)
atu_single_hc.setName("ATU SingleDuctVAV reheat coil")
pl.addDemandBranchForComponent(atu_single_hc)
atu_single = AirTerminalSingleDuctVAVReheat.new(m, m.alwaysOnDiscreteSchedule, atu_single_hc)
atu_single.setName("ATU SingleDuctVAVReheat")

# True, as it should be
atu_single_hc.containingHVACComponent.is_initialized

# Create an ATU FourPipeInduction
atu_fourpipe_hc= CoilHeatingWater.new(m)
atu_fourpipe_hc.setName("ATU FourPipe Heating coil")
pl.addDemandBranchForComponent(atu_fourpipe_hc)
atu_fourpipe = AirTerminalSingleDuctConstantVolumeFourPipeInduction.new(m, atu_fourpipe_hc)
atu_fourpipe.setName("ATU FourPipeInduction")

# That's it: this returns false
atu_fourpipe_hc.containingHVACComponent.is_initialized

# Clone SingleDuct
atu_single_clone = atu_single.clone(m).to_AirTerminalSingleDuctVAVReheat.get
atu_single_clone_hc = atu_single_clone.reheatCoil.to_CoilHeatingWater.get
# False
atu_single_clone_hc.plantLoop.is_initialized

# Clone ATU FourPipe
atu_fourpipe_clone = atu_fourpipe.clone(m).to_AirTerminalSingleDuctConstantVolumeFourPipeInduction.get
atu_fourpipe_clone_hc = atu_fourpipe_clone.heatingCoil.to_CoilHeatingWater.get
# False
atu_fourpipe_clone_hc.plantLoop.is_initialized

```

**Comment by jmarrec** [comment]
I then realized that ATU:FourPipeInduction is special because it has both a cooling and heating coil.
ATU:CooledBeam is also special because the coilCoolingCooledBeam is a StraightComponent and not a WaterToAirComponent, so I add to implement a check for this in AirLoopHVAC::addBranchForZone.

Wrote Gtests for all terminal types to make sure "magic" is indeed happening.

**Comment by jmarrec** [comment]
Fixed in #2854
